### PR TITLE
[ Fabric ]: Fix for chaincode install job

### PR DESCRIPTION
**Pull request by Oreo-Tech**
**Changelog**
- Fix

Description: On line number 158 we are doing change directory to "$GOPATH/src/github.com/chaincode" but on the next command (line no: 160) we were expecting the current directory to be relative to "github.com/chaincode/${CHAINCODE_NAME}/${CHAINCODE_MAINDIR}" path to install the chaincode which is not reachable from the present directory as we already moved at $GOPATH/src/github.com/chaincode/ and the install command is not able to find the chaincode to install
 

**Reviewed by**
@jagpreetsinghsasan 

 

**Linked issue**
#1166

### CODE REVIEW

**Comment by sownak** [comment]
Thanks for the PR. The sign-off is missing. Also has this been tested? Because I think without the main file, the install will not be successful.

**Review by sownak — CHANGES_REQUESTED** [review]
Signoff needs to be added.

**Comment by jagpreetsinghsasan** [comment]
I don't think this will work as the chaincode folder doesn't necessarily have the main file to run.
(For example, fabcar java cc)

**Comment by Oreo-Tech** [comment]
The Fix was majorly in focusing on providing the correct relative path where chaincode is stored "$GOPATH/src/github.com/chaincode" instead of the earlier coded path "github.com/chaincode.." which was not reachable as we are already present in the folder due to "cd $GOPATH/src/github.com/chaincode" instruction on line 158

talking about the removal of "${CHAINCODE_NAME}/${CHAINCODE_MAINDIR}" section from the code, I agree its missing but do we really need mentioning a Main directory which does not play any major role as we already have "Values.chaincode.repository.path " doing the work to provide path for main file to us

Line - 150: cp -R /tmp/chaincode/{{ $.Values.chaincode.repository.path }}/* $GOPATH/src/github.com/chaincode/

all we do is track the chaincode repository path and copy it into local path so I thought to avoid the added dependency of main dir and provide the correct repository path

**Comment by sownak** [comment]
Yes we do need mentioning a Main directory because without that, the chaincode does not install. In your case, the main directory may be the same as the chaincode directory, but that may not be true for everyone else.

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml [hunk 1]
```diff
@@ -157,7 +157,7 @@ spec:
           mkdir -p $GOPATH/bin && curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
           cd $GOPATH/src/github.com/chaincode && dep ensure
           ## Installing chaincode on a peer
-          peer chaincode install -n ${CHAINCODE_NAME} -v ${CHAINCODE_VERSION} -p github.com/chaincode/${CHAINCODE_NAME}/${CHAINCODE_MAINDIR}
+          peer chaincode install -n ${CHAINCODE_NAME} -v ${CHAINCODE_VERSION} -p $GOPATH/src/github.com/chaincode
```
Comment (sownak) [inline platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:160]: What if there are multiple chaincodes in the $GOPATH/src/github.com/chaincode/ directory? How will the install chaincode work then? And each chaincode can have a different main directory, as I mentioned in the comment.

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml (hunk 1 above)
Comment (bafsunsu) [reply platforms/hyperledger-fabric/charts/install_chaincode/templates/install_chaincode.yaml:160]: thanks, got your point, am making the changes accordingly and will update it

File: platforms/hyperledger-fabric/configuration/roles/create/chaincode/install/tasks/valuefile.yaml [hunk 2]
```diff
@@ -45,7 +45,7 @@
     type: "install_chaincode_job"
     peer_name: "{{ peer.name }}"
     peer_address: "{{ peer.name }}.{{ namespace }}:{{ peer.grpc.port }}"
-    component_name: "chaincode-install-{{ name }}-{{ peer.name }}-{{ peer.chaincode.version | replace('.','-')}}"
+    component_name: "chaincode-install-{{ name | lower }}-{{ peer.name }}-{{peer.chaincode.name}}{{peer.chaincode.version}}"
```
Comment (arsulegai) [inline platforms/hyperledger-fabric/configuration/roles/create/chaincode/install/tasks/valuefile.yaml:48]: Please add back `| replace('.','-')` to the version.

**Review by arsulegai — CHANGES_REQUESTED** [review]

**Comment by suvajit-sarkar** [comment]
closing this PR as it has gone stale, if required create a new PR to address the issue

**Pull request closed without merging**